netutils/platform_mapper.py

```diff
@@ -0,0 +1,231 @@
+            self.build = re_main_minor_type_build.group(4)  # type: ignore[union-attr]
+
+            # Set empty params for service pieces and complete them if a second indice exists from the version split
+            # Define isservice, isfrs, isspecial, ismaintenance
+            self.service = None
+            self.service_build = None
+            self.service_respin = None
+            if len(split_version) > 1:
+                re_service_respin = re.search(r"([sSdD])?(\d+)?(\.)?(\d+)?", split_version[1])
+                self.service = re_service_respin.group(1)  # type: ignore[union-attr]
+                self.service_build = re_service_respin.group(2)  # type: ignore[union-attr]
+                self.service_respin = re_service_respin.group(4)  # type: ignore[union-attr]
+
+                if self.service.lower() == "s":
+                    self.isservice = True
+
+                # Juniper looks at the D in special releases like it's the R in normal releases; Use it as the frs identifier
+                elif self.service.lower() == "d":
+                    if self.service_build is None or int(self.service_build) <= 1:
+                        self.isfrs = True
+
+            if self.type is not None:
+                if self.type.lower() == "x":
+                    self.isspecial = True
+                elif self.type.lower() == "s":
+                    self.isservice = True
+
+                if self.type.lower() == "r" and hasattr(self, "build"):
+                    if self.build is None or int(self.build) <= 1:
+                        self.isfrs = True
+                    else:
+                        self.ismaintenance = True
+
+    def get_nist_urls(self, api_key: str) -> t.List[str]:
+        """Create a list of possible NIST Url strings for JuniperPlatform.
+
+        Args:
+            api_key: NIST-API-KEY - Request here https://nvd.nist.gov/developers/request-an-api-key
+
+        Returns:
+            List of NIST CPE URLs that may contain platform data.
+
+        Examples:
+            >>> JuniperPlatform('junos','12.1R3-S4.3').get_nist_urls('YOURKEY')
+            ['https://services.nvd.nist.gov/rest/json/cpes/1.0?apiKey=YOURKEY&addOns=cves&cpeMatchString=cpe:2.3:o:juniper:junos:12.1R3:S4.3:*:*:*:*:*:*', 'https://services.nvd.nist.gov/rest/json/cpes/1.0?apiKey=YOURKEY&addOns=cves&cpeMatchString=cpe:2.3:o:juniper:junos:12.1R3-S4.3:*:*:*:*:*:*:*']
+
+            >>> JuniperPlatform('junos','12.1').get_nist_urls('YOURKEY')
+            ['https://services.nvd.nist.gov/rest/json/cpes/1.0?apiKey=YOURKEY&addOns=cves&cpeMatchString=cpe:2.3:o:juniper:junos:12.1:-:*:*:*:*:*:*']
+        """
+        nist_urls = []
+        base_url = f"""https://services.nvd.nist.gov/rest/json/cpes/1.0?apiKey={api_key}&addOns=cves&cpeMatchString=cpe:2.3:o:juniper:junos"""
+        base_extension = f"{base_url}:{self.main}.{self.minor}"
+
+        # X Series (Special) Examples: 12.1x47:d40, 12.2x50:d41.1
+        if self.isspecial:
```

Comment:
seems like a ton of branches here.  Can we simplify it a bit?

Reply:
This was simplified more and re-committed.

Resolving this conversation but the new code is being discussed below.